ejb3/src/main/java/org/jboss/as/ejb3/remote/AssociationImpl.java

```diff
@@ -73,23 +73,59 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.CancellationException;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 import java.util.concurrent.Future;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
 
 /**
  * @author <a href="mailto:tadamski@redhat.com">Tomasz Adamski</a>
  */
-final class AssociationImpl implements Association {
+final class AssociationImpl implements Association, Registry.Listener<String, List<ClientMapping>>, AutoCloseable {
```

Comment:
I'd prefer to stick with composition for the listener instead of inheritance in this particular case.  In the general case, it's important to separate structure refactors from logic changes in PRs because otherwise it's not clear what the actual change is.